### PR TITLE
Spice manager global

### PR DESCRIPTION
**Pull request by nicHoch**
- removed time and position spice context
- combined into a single "Spice" class
- not a context anymore as this has to be propagated to the sub processes for parallel computing 
- the global singleton will be auto instantiated for testing but should be set up manually for any __main__

### CODE REVIEW

File: stixcore/processing/L0toL1.py [hunk 1]
```diff
@@ -74,6 +81,10 @@ def process_type(files, processor):
     fits_path = Path('/home/shane/fits_test/L0')
     bd = Path('/home/shane/fits_test/')
 
+    # possible set an alternative spice kernel if not the latest should be used
+    SpiceKernelManager.instance = SpiceKernelManager(Path(CONFIG.get("Paths", "spice_kernels")))
```
Comment (samaloney) [inline stixcore/processing/L0toL1.py:85]: Does this need to be a singleton?

**Review by samaloney — APPROVED** [review]

**Comment by samaloney** [comment]
Also just wonder how this will work with IDL will you give it the path or set it when calling from hissw?

File: stixcore/processing/L0toL1.py (hunk 1 above)
Comment (nicHoch) [reply stixcore/processing/L0toL1.py:85]: hmm right the manager do not really have to be a singleton.

just the Spice object need to be one if we will avoid the pass it around everywhere 

will fix this

**Comment by nicHoch** [comment]
> Also just wonder how this will work with IDL will you give it the path or set it when calling from hissw?

We can decide if we want to pass the path into hissw or use the environment variable. both ways are ready to use.

**Pull request merged**